### PR TITLE
LPS-159834 [Raylife AP] Creating PRODUCT PERFORMANCE Basic Chart Structure

### PR DESCRIPTION
**Pull request by eduardo-diniz**

### CODE REVIEW

**Comment by silasbsi** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
Just starting reviewing :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 1]
```diff
@@ -28,36 +29,147 @@ import {
...
+			label.index < actualMonth + 1 && label.index > actualMonth - 6
+	)
+	.map((label: MonthProperties) => label.label);
+
+const setLabelThree = Object.values(dataColumn)
+	.filter(
+		(label: MonthProperties) =>
+			label.index < actualMonth + 1 && label.index > actualMonth - 3
+	)
+	.map((label: MonthProperties) => label.label);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx:121]: Same here, you should try something like: 
```suggestion
function setLabelByPeriod (filteredPeriod) {
	const definedLabel = filteredPeriod.map((label: MonthProperties) => label.label);

	return definedLabel;
}
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 2]
```diff
@@ -28,36 +29,147 @@ import {
...
+
+const paddingValue = 100;
+
 const ProductPerformance = () => {
 	const [products, setProducts] = useState<ProductCell[]>([]);
-	const [timePeriod, setTimePeriod] = useState(TIME_PERIODS[0]);
+	const [timePeriod, setTimePeriod] = useState(PERIOD.THREE_MONTH);
+	const [filt, setFilt] = useState<MonthProperties[]>(filterYearly);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx:128]: What does "filter" do? Try using a more specific name.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 3]
```diff
@@ -28,36 +29,147 @@ import {
...
+	const achieved = filt.map((item: MonthProperties) =>
+		item.achieved > item.goals ? item.goals : item.achieved
+	);
+
+	const exceeded = filt.map((item: MonthProperties) =>
+		item.achieved > item.goals ? item.achieved - item.goals : NaN
+	);
+	const goals = filt.map((item: MonthProperties) =>
+		item.goals < 0 || item.goals < item.achieved ? NaN : item.goals
+	);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx:141]: Same here, try to use a generic function.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 4]
```diff
@@ -28,36 +29,147 @@ import {
...
+			setFilt(filterThree);
+			ref.current.categories(setLabelThree);
+		}
+		if (timePeriod === PERIOD.YTD) {
+			setFilt(filterYearly);
+			ref.current.categories(setLabelYearly);
+		} else {
+			timePeriod = PERIOD.YTD;
+		}
+	};
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx:172]: Same here.

**Review by joanagrego — CHANGES_REQUESTED** [review]
Nice job! I added some suggestions :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 5]
```diff
@@ -28,36 +29,147 @@ import {
...
-	initialReferenceDate?: string;
-	productExternalReferenceCode: string;
+const colors: {[keys: string]: {}} = {
+	achieved: '#55C2FF',
+	exceeded: '#FFD76E',
+	goals: '#DCF1FD',
 };
 
+const date = new Date();
+const actualMonth = date.getMonth();
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 6]
```diff
@@ -28,36 +29,147 @@ import {
...
+
+const filterSix = Object.values(dataColumn).filter(
+	(month: MonthProperties) =>
+		month.index < actualMonth + 1 && month.index > actualMonth - 6
+);
+
+const filterThree = Object.values(dataColumn).filter(
+	(month: MonthProperties) =>
+		month.index < actualMonth + 1 && month.index > actualMonth - 3
+);
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx (hunk 1 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx (hunk 2 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx (hunk 3 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx]: Will refractory in next sub-task.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx (hunk 4 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx]: Will refractory in next sub-task.

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by JoMaAlves** [comment]
ci:test:sf

**Review by joanagrego — APPROVED** [review]
Nice job! :white_check_mark:

**Comment by eliasantosf** [comment]
ci:test:sf

**Review by eliasantosf — APPROVED** [review]
LGTM :fire: Thanks @eduardo-diniz

**Comment by silasbsi** [comment]
ci:forward

**Comment by pedroliveirav** [comment]
good job!

**Pull request closed without merging**